concepts/0017-attachments/README.md

```diff
@@ -338,6 +336,72 @@ whichever mechanism(s) are best suited to its individual needs and capabilities.
 [TODO: discuss sending an empty message with just attachments, and how to
 request a send of an attachment, or an alternate download method for it]
 
+### Signing Attachments
+
+In some cases it may be desirable to sign an attachment in addition to or instead of
+signing the message as a whole. Consider a home-buying protocol; the home inspection
+needs to be signed even when it is removed from a messaging flow. Attachments may
+also be signed by a party separate from the sender of the message, or using a different
+signing key when the sender is performing key rotation.
+
+Embedded and appended attachments support signatures by the addition of a `jws` field
+containing a signature in [JWS (RFC 7515) format](https://tools.ietf.org/html/rfc7515)
+with [Detached Content](https://tools.ietf.org/html/rfc7515#appendix-F).
+The payload of the JWS is the raw data of the attachment, whether externally referenced
+or encoded in base64 format, and is not contained within the signature itself.
+Signatures over inlined JSON attachments are not currently defined as this
+depends upon a canonical serialization for the data.
+
+Sample JWS-signed attachment:
+
+```jsonc
+{
+  "@type": "https://didcomm.org/xhomebuy/1.0/home_insp",
+  "inspection_date": "2020-03-25",
+  "inspection_address": "123 Villa de Las Fuentes, Toledo, Spain",
+  "comment": "Here's that report you asked for.",
+  "report~attach": {
+    "mime-type": "application/pdf",
+    "filename": "Garcia-inspection-March-25.pdf",
+    "data": {
+      "base64": "eyJ0eXAiOiJKV1QiLA0KICJhbGciOiJIUzI1NiJ... (bytes omitted to shorten)",
```

Comment:
I think the JWS should be inside "data", not a peer of it. That's because "data" already supports "hash"; "jws" is a close analog to "hash".

Reply:
I wondered about that, and had it in `data` originally but it was suggested on the Aries call to move it up a level. I agree it makes sense to have them at the same level, whether inside or outside the data block.

Reply:
I don't think this is a signature over the mime type, the filename, etc -- only over the raw bytes of the data. So I think it belongs next to the raw bytes of the data, not at the outer level.

Reply:
I prefer the attachment level, but not strongly. I think of it as a 'signed attachment' not an 'attachment with signed data'.

Reply:
My $0.02CDN is that is should be within the `data` element as @dhh1128 suggests.